src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -71,7 +71,31 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
     public ResponseEntity<StudentExam> getStudentExam(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long studentExamId) {
         log.debug("REST request to get student exam : {}", studentExamId);
         Optional<ResponseEntity<StudentExam>> accessFailure = examAccessService.checkCourseAndExamAndStudentExamAccess(courseId, examId, studentExamId);
-        return accessFailure.orElseGet(() -> ResponseEntity.ok(studentExamService.findOneWithExercises(studentExamId)));
+
+        StudentExam studentExam = studentExamService.findOneWithExercises(studentExamId);
+        // we reload the quiz exercise because we also need the quiz questions and it is not possible to load them in a generic way with the entity graph used in
```

Comment:
Might want to declare `List<Exercise> exercises = studentExam.getExercises()` here to avoid multiple database accesses if that is not specifically intended

Reply:
there is no database access, all exercises are already fetched.